cli/azd/pkg/environment/manager.go

```diff
@@ -221,6 +168,104 @@ func (m *manager) LoadOrCreateInteractive(ctx context.Context, environmentName s
 	return env, nil
 }
 
+func (m *manager) loadOrInitEnvironment(ctx context.Context, environmentName string) (*Environment, bool, error) {
+	// If there's a default environment, use that
+	if environmentName == "" {
+		var err error
+		environmentName, err = m.azdContext.GetDefaultEnvironmentName()
+		if err != nil {
+			return nil, false, fmt.Errorf("getting default environment: %w", err)
+		}
+	}
+
+	if environmentName != "" {
+		env, err := m.Get(ctx, environmentName)
+		switch {
+		case errors.Is(err, ErrNotFound):
+			msg := fmt.Sprintf("Environment '%s' does not exist, would you like to create it?", environmentName)
+			shouldCreate, promptErr := m.console.Confirm(ctx, input.ConsoleOptions{
+				Message:      msg,
+				DefaultValue: true,
+			})
+			if promptErr != nil {
+				return nil, false, fmt.Errorf("prompting to create environment '%s': %w", environmentName, promptErr)
+			}
+			if !shouldCreate {
+				return nil, false, fmt.Errorf("environment '%s' not found: %w", environmentName, err)
+			}
+		case err != nil:
+			return nil, false, fmt.Errorf("loading environment '%s': %w", environmentName, err)
+		case err == nil:
+			return env, false, nil
+		}
+	}
+
+	// Two cases if we get to here:
+	// - The user has not specified an environment name, and there was no default environment set
+	// - The user has specified an environment name, but the named environment didn't exist and they told us they would
+	//   like us to create it.
+	if environmentName != "" && !IsValidEnvironmentName(environmentName) {
+		fmt.Fprintf(
+			m.console.Handles().Stdout,
+			"environment name '%s' is invalid (it should contain only alphanumeric characters and hyphens)\n",
+			environmentName)
+		return nil, false, fmt.Errorf(
+			"environment name '%s' is invalid (it should contain only alphanumeric characters and hyphens)",
+			environmentName)
+	}
+
+	// No environment name, no default environment set.
+	// Ask the user if they want to create a new environment or select an existing one
+	if environmentName == "" {
+		envs, err := m.List(ctx)
+		if err != nil {
+			return nil, false, err
+		}
+
+		// Selection, 0 is the option to create a new environment
+		selection := 0
+		choices := make([]string, 0, len(envs)+1)
+		choices = append(choices, "Create a new environment")
+		if len(envs) > 0 {
+			for _, env := range envs {
+				choices = append(choices, env.Name)
+			}
+
+			selection, err = m.console.Select(ctx, input.ConsoleOptions{
+				Message: "Select an environment to use:",
+				Options: choices,
+			})
```

Comment:
If we are expecting to fail here when `--no-prompt` is used, we might want to have a comment about it.

Reply:
I do really appreciate the callout though, we definitely need to think more about different ways of interacting with the CLI: prompt/flag/env-var.